queue/queue_test.go

```diff
@@ -598,6 +704,86 @@ func TestUtube_Put(t *testing.T) {
 	}
 }
 
+func TestTask_Touch(t *testing.T) {
+	conn := test_helpers.ConnectWithValidation(t, server, opts)
+	defer conn.Close()
+
+	tests := []struct {
+		name string
+		cfg  queue.Cfg
+		ok   bool
+	}{
+		{"test_queue",
+			queue.Cfg{
+				Temporary: true,
+				Kind:      queue.FIFO,
+			},
+			false,
+		},
+		{"test_queue_ttl",
+			queue.Cfg{
+				Temporary: true,
+				Kind:      queue.FIFO_TTL,
+				Opts:      queue.Opts{Ttl: 5 * time.Second},
+			},
+			true,
+		},
+		{"test_utube",
+			queue.Cfg{
+				Temporary: true,
+				Kind:      queue.UTUBE,
+			},
+			false,
+		},
+		{"test_utube_ttl",
+			queue.Cfg{
+				Temporary: true,
+				Kind:      queue.UTUBE_TTL,
+				Opts:      queue.Opts{Ttl: 5 * time.Second},
+			},
+			true,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			var task *queue.Task
+
+			q := createQueue(t, conn, tc.name, tc.cfg)
+			defer func() {
+				if task != nil {
+					if err := task.Ack(); err != nil {
+						t.Fatalf("Failed to Ack: %s", err)
+					}
+				}
+				dropQueue(t, q)
+			}()
+
+			putData := "put_data"
+			_, err := q.PutWithOpts(putData,
+				queue.Opts{
+					Ttl:   10 * time.Second,
+					Utube: "test_utube",
+				})
+			if err != nil {
+				t.Fatalf("Failed put a task: %s", err)
+			}
+
+			task, err = q.TakeTimeout(2 * time.Second)
+			if err != nil {
+				t.Fatalf("Failed to take task from utube: %s", err)
+			}
+
+			err = task.Touch(1 * time.Second)
```

Comment:
Is there any other thing we could check here? Or there is no way to conveniently extract "time remained" var and add some assert for it?

Reply:
Unfortunately, there is no normal way, only timeouts and sleeps in tests. I don't like this approach, it increases the execution time of tests and makes them fragile.
 
 Maybe a response from the server without errors will be enough in this case?

Reply:
Let it be like it is now